pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -110,6 +118,49 @@ func pushToShifuTelemetryCollectionService(message *http.Response, request *v1al
 	return nil
 }
 
+func getSecret(c *rest.RESTClient, name, ns string) (map[string]string, error) {
+	secret := v1.Secret{}
+	secretPath := fmt.Sprintf("/api/v1/namespaces/%s/%s/%s", ns, SecretResource, name)
+	if err := c.Get().AbsPath(secretPath).Do(context.TODO()).Into(&secret); err != nil {
+		return nil, err
+	}
+	res := make(map[string]string)
+	for k, v := range secret.Data {
+		res[k] = string(v)
+	}
+	return res, nil
+}
+
+func injectSecret(c *rest.RESTClient, ts *v1alpha1.TelemetryService, ns string) {
+	if ts.Spec.ServiceSettings == nil {
+		logger.Warn("empty telemetry service setting.")
+		return
+	}
+	if ts.Spec.ServiceSettings.HTTPSetting == nil {
+		logger.Info("service setting is not HTTP, skip secret injection")
+		return
+	}
+	// we use the telemetry name to find the secret
+	secret, err := getSecret(c, ts.Name, ns)
+	if err != nil {
+		logger.Errorf("unable to get secret for telemetry %v, error: %v, use plaintext password from telemetry setting", ts.Name, err)
+		return
+	}
+	var exist bool
+	*ts.Spec.ServiceSettings.HTTPSetting.Password, exist = secret[PasswordSecretField]
+	if !exist {
+		logger.Errorf("the %v field not found in telemetry secret", PasswordSecretField)
+	} else {
+		logger.Info("HTTPSetting.Password load from secret")
+	}
+	*ts.Spec.ServiceSettings.HTTPSetting.Username, exist = secret[UsernameSecretField]
+	if !exist {
+		logger.Errorf("the %v field not found in telemetry secret", UsernameSecretField)
+	} else {
+		logger.Info("HTTPSetting.Username load from secret")
+	}
+}
+
 func getTelemetryCollectionServiceMap(ds *DeviceShifuBase) (map[string]v1alpha1.TelemetryServiceSpec, error) {
```

Comment:
Please add some comment to this function, it's not very easy to understand.

Reply:
ok